### ISSUE TITLE
Snippets expand mid word

### ISSUE THREAD

**Issue by tomasklaen** [microsoft/vscode#138846]
- VS Code Version: 1.63.0
- OS Version: Win 10
- ✔️ also happens on insiders with no extensions

vscode expands snippets even when they are a part of a different word. This started with the most recent update.

Steps to Reproduce:

Lets say I have a snippet **`bar`**.

I write `foobar|` and press tab to autocomplete.

vscode will expand the snippet `bar` even though it's a part of a different word.

It even ignores capitalization, and will expand snippet `BAR` when cursor is next to a word `bar|`.

**Comment by gjsjohnmurray** [comment]
Maybe #138707

Are you able to test with today's Insiders?

**Comment by tomasklaen** [comment]
This is on windows portable (zip package) insiders build I've downloaded when creating this issue an hour ago. Tried again now, and there is no newer package available.

But it's weird. It sometimes works as intended, sometimes doesn't and expands midword.

If I try the example from opening post in a new file with typescript language selected, vscode will expand **ar** from `foobar|` to **Async Function Expression** snippet, resulting in:

```ts
foobasync (params:type) => {
    
}
```

I have a lot of snippets, and needles to say the autocomplete is so broken since the last update it disrupted my workflow to the point where I stopped pressing tab completely :( 9 out of 10 times it doesn't autocomplete, but expands a snippet I didn't want.
